File: Factorial.java
import java.util.Scanner;

public class Factorial {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int num;
        System.out.println("enter the number:");
        num = in.nextInt();
        System.out.println("the factorial of " + num + " is " + factorial(num));
    }

    static int factorial(int num) {
        if (num == 0 || num == 1) {
            return 1;
        } else {
            return num * (factorial(num - 1));
        }
    }
}

File: Factorial.c
#include<stdio.h>
int factorial(int num)
{ 
    if(num==0 || num==1 )
    return 1;
    else
    return num*(factorial(num-1));
}
int main()
{ int num;
    printf("enter the number:");
    scanf("%d",&num);
    printf("the factorial of %d is: %d",num,factorial(num));
    return 0;
}